### PR TITLE
Guaranty local IMAP session consistency

### PR DESCRIPTION
**Pull request by hoangdat**

### CODE REVIEW

File: mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/EventDispatcher.java [hunk 1]
```diff
@@ -110,6 +100,28 @@ void start() {
...
+
+        Flux<MailboxListener> localGroupListener = Flux.fromIterable(groupRegistrationHandler
+            .getGroupRegistrations().values())
+            .map(groupRegistration -> groupRegistration.getMailboxListener());
+
+        return Flux.concat(localKeyListener, localGroupListener)
+            .subscribeOn(Schedulers.elastic())
+            .filter(mailboxListener -> mailboxListener.getExecutionMode() == MailboxListener.ExecutionMode.SYNCHRONOUS)
+            .flatMap(mailboxListener -> Mono.fromRunnable(() -> {
+                try {
```
Comment (chibenwa) [inline mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/EventDispatcher.java]: Can this be extracted in another method?

**Comment by hoangdat** [comment]
Sometimes, I failed in test `unregisteredGroupListenerShouldNotReceiveEvents()` on `EventBusContract.java`. I found that, when `unregister` a `GroupRegistration`, `receiver.close()` but the consumer still executed. Anybody have a solution for this.

**Comment by chibenwa** [comment]
> Sometimes, I failed in test unregisteredGroupListenerShouldNotReceiveEvents() on EventBusContract.java. I found that, when unregister a GroupRegistration, receiver.close() but the consumer still executed. Anybody have a solution for this.

It looks like an isolation problem coming from a non closed receiver. @trantienduchn did have a commit for this. Rebasing on the latest version of https://github.com/linagora/james-project/pull/2070 should certainly help.

**Comment by chibenwa** [comment]
Was :green_apple: 

test this please

**Comment by aduprat** [comment]
Was :green_apple: before restarting the CI.

**Comment by openpaas-james** [comment]
Build triggered. sha1 is merged.

**Comment by openpaas-james** [comment]
Build started sha1 is merged.

**Comment by openpaas-james** [comment]
Build finished. No test results found.

**Pull request closed without merging**